test/org/javarosa/core/model/instance/TestHelpers.java

```diff
@@ -0,0 +1,38 @@
+package org.javarosa.core.model.instance;
+
+import static org.javarosa.xpath.expr.XPathPathExpr.INIT_CONTEXT_RELATIVE;
+import static org.javarosa.xpath.expr.XPathPathExpr.INIT_CONTEXT_ROOT;
+import static org.javarosa.xpath.expr.XPathStep.ABBR_PARENT;
+import static org.javarosa.xpath.expr.XPathStep.ABBR_SELF;
+import static org.javarosa.xpath.expr.XPathStep.AXIS_CHILD;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.javarosa.xpath.expr.XPathPathExpr;
+import org.javarosa.xpath.expr.XPathQName;
+import org.javarosa.xpath.expr.XPathStep;
+
+public class TestHelpers {
+    static TreeReference buildRef(String xpath) {
```

Comment:
I was wondering whether we might already have a way to do this and I think `XPathReference(String)` might be just that.

Reply:
Ugh! I'm a dummy! That's much better, thanks!

Reply:
Using `new XPathReference(String nodeset)` works correctly but to cover all the paths of `parent()`, I need to produce a base xpath that would pass this check in `TreeReference.java` line 250:
```java
!baseReference.isAbsolute() && baseReference.size() == 0
```

This is consistent with an empty `""` xpath, right?

In my first implementation of `TestHelpers.buildRef(String xpath)` this xpath turned to a relative `TreeReference` with an empty list of steps, which would pass the check above.

If I do `new XPathReference("").getReference()` I get an `org.javarosa.xpath.XPathException: XPath evaluation: Parse error in XPath path: [].` error.

What now?
- I can remove the test case and leave that line not covered by any test
- We can come up with another xpath that won't fail with the `XPathPreference` class' constructor that passes the check on line 250. **I'd need some help here.**
- I can conditionally run my old `TestHelpers.buildRef(String xpath)` just for the `""` xpath case

Reply:
After discussing this with @lognaturel, we have confirmed that the interpretation of what an empty xpath should be as a `TreeReference` is correct: a relative TreeReference with an empty list of steps.

We also have agreed on conditionally running the old `TestHelpers.buildRef(String xpath)` to avoid the `XPathException` when we get an empty xpath.

This means that there's no need to add more changes regarding this.